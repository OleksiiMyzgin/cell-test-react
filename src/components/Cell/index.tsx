import classes from './styles.module.css';

interface CellProps {
  cell: number;
}

function Cell({ cell }: CellProps) {
  const color = cell ? 'black' : 'white';
  return <div className={`${classes.cell} ${classes[color]}`} />;
}

export default Cell;
