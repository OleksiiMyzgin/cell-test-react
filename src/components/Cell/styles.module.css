.cell {
  width: 8px;
  height: 8px;
  border: 1px solid grey;
}

.black {
  background-color: black;
}

.white {
  background-color: white;
}
