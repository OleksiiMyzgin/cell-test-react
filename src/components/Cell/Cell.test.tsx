import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Cell from './index';

describe('Cell tests:', () => {
  test('snapshot', () => {
    const component = renderer.create(<Cell cell={0} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('display white element', () => {
    const { container } = render(<Cell cell={0} />);

    expect(container.querySelector('.cell')).toHaveClass('cell white');
  });

  test('display black element', () => {
    const { container } = render(<Cell cell={1} />);

    expect(container.querySelector('.cell')).toHaveClass('cell black');
  });
});
