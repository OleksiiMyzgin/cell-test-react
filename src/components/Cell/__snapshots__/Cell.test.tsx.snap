// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell tests: snapshot 1`] = `
<div
  className="cell white"
/>
`;
